### DISCUSSION TITLE
Plugins

### DISCUSSION THREAD

**Discussion by LouisHaftmann** [trpc/trpc#1742] [Ideas]
```tsx
const plugin = router<any, { meta2: boolean }>().middleware(({ next }) => {
  return next();
});

router<any, { meta1: string }>().plugin(plugin)
```

We would need a function (`router.plugin()`) similar to `router.merge()` that also merges context and meta

**Comment by jlalmes** [comment]
Recommend taking a look at the v10 playground: https://github.com/trpc/v10-playground

**Reply by KATT** [reply]
In particular https://github.com/trpc/v10-playground/pull/33 which is the most likely format we'll land on.

Once we've landed on a good way forward, I would love some help on getting this implemented [with backwards compatability]

**Reply by LouisHaftmann** [reply]
v10 is looking amazing already, good job so far

really looking forward to working on this
